kernel-ewasm/cap9-std/src/syscalls.rs

```diff
@@ -145,6 +118,31 @@ impl SysCallAction {
                 }
                 false
             },
+            // Register Procedure syscall
+            SysCallAction::Register(RegisterProc{proc_id,address:_, cap_list}) => {
+                // Check that this procedure has the correct capability to
+                // register a procedure of the given key.
+                if let Capability::ProcedureRegister(proc_table::cap::ProcedureRegisterCap {prefix, key}) = cap {
```

Comment:
If `cap` isn't a `ProcedureReg` it should return `false` here. I would advise defaulting to returning false on most cases, just so we don't miss out on cases like this.

Reply:
No you're not being pedantic I've just made a massive typo. You'll notice all the rest have a false fallthrough. Thanks for catching that. The blocks are also incorrectly nested.

Reply:
Fixed with a3ce76a.